### PR TITLE
WIP: Add support for a app configuration button in the sidebar

### PR DESCRIPTION
**Pull request by robbiet480**
This PR adds a new button to the sidebar that when tapped will fire a JS event which can be caught by a external application to cause its native settings view to open.

image

[Here's a little video too](https://cdn.discordapp.com/attachments/351047592588869643/569047305936699393/video.mov)

### CODE REVIEW

**Comment by robbiet480** [comment]
@balloob Can you outline how exactly you want me to show this button only if something is listening for it? Actually, we could do it that simply: Only show button if someone is listening for event? Idk.

File: src/components/ha-sidebar.ts [hunk 1]
```diff
@@ -117,6 +117,17 @@ class HaSidebar extends LitElement {
             </a>
           `
         )}
+        <a href="#external-app-configuration" tabindex="-1">
```
Comment (balloob) [inline src/components/ha-sidebar.ts]: No need to use an `a` tag if we're not using it as an `<a>` ?

**Comment by balloob** [comment]
I think that we should add a way for the frontend to know that we run in external auth mode. 

We can do that by adding stuff to the auth class defined in `external_auth.ts`, as it's accessible as `hass.auth` via the app. 

In `src/types.ts`, define a new interface that extends auth:

```ts
interface AuthWithExternal extends Auth {
  external?: ExternalInterface
}
```

Now define an external interface that allows querying for a config object, for now the only config option will be if we should show a sidebar menu. But we will also use the external object to route haptics back. 

Now in src/types.ts, update `HomeAssistant` to change type of `auth` from `Auth` to `AuthWithExternal`. Now inside the sidebar you can query for it.

**Comment by balloob** [comment]
I feel like I should intervene and open a PR myself to add my suggested interface ;)

**Comment by robbiet480** [comment]
>I feel like I should intervene and open a PR myself to add my suggested interface ;)

@balloob yes you should :)

File: src/components/ha-sidebar.ts (hunk 1 above)
Comment (robbiet480) [reply src/components/ha-sidebar.ts]: Without it the button doesn’t match the other buttons.

**Comment by robbiet480** [comment]
If we're making this general system we should change it to `external_app=1` instead since it signals more than just auth. Or, should we just have another parameter `external_app` in addition to `external_auth` to let the app specific stuff be opt in?

File: src/components/ha-sidebar.ts (hunk 1 above)
Comment (robbiet480) [reply src/components/ha-sidebar.ts]: Correction, just checked again. It looks fine but doesn't behave the same. Clicking it leaves it highlighted until clicked away.

image

**Comment by balloob** [comment]
Not going to change the parameter, since it's being used already. Having an external interface attached to external auth is good, as it means people can only use it if they also provide external auth, meaning that we can enforce a good user experience.

**Comment by balloob** [comment]
Merged this PR into #3112. Removed the connected events. If we want those, open a new PR and add a use case. Also consolidate in 1 event, like `connection-status`

**Pull request closed without merging**